### ISSUE TITLE
Enable Logins for a Password Pusher Deployment

### ISSUE THREAD

**Issue by Abhishek0237** [pglombardo/PasswordPusher#2307]
I tried to Enable sign in but i’m getting 500 error when doing sign up..i’m using dcoker compose configuration both my password pusher and database running on the docker containers.can you please help me

**Comment by pglombardo** [comment]
Hi @Abhishek0237 - it's most likely a misconfiguration with the email settings.  Could you run the validations documented here?

https://github.com/pglombardo/PasswordPusher/wiki/How-to-Validate-and-Test-Email-Settings

**Comment by Abhishek0237** [comment]
Hi @pglombardo I will do the same and let you .
Thanks for your response ☺️

**Comment by Abhishek0237** [comment]
Hi @pglombardo Thanks for help it working now☺️🙌🏼

My other question whenever user sign into their account and try to share password instead of generating the password link and sharing with us  is it not possible we admins are directly get notification or password link on our smtp email id whenever users generate the password..

**Comment by pglombardo** [comment]
Hi @Abhishek0237 - excellent on email working now.

As for the second question, I apologize but I don't understand.  Could you rephrase it?

**Comment by Abhishek0237** [comment]
Hello, @pglombardo My intended meaning was when a user logs into Password Pusher and want to distribute their password. Is it feasible that when they establish a password, we will automatically receive a link in our email and we will only receive notices by email that the user has made a password?

**Comment by pglombardo** [comment]
Ah ok I get it.  We don't have application wide activity notifications currently.

Alternatively you can can log into the [Admin Dashboard](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md#admin-dashboard) and check users and their activity from time to time.

Does this help in what you want to do?

**Comment by Abhishek0237** [comment]
Thanks for the information 

Is it not possible to check the password shared by user from Admin dashboard?

**Comment by Abhishek0237** [comment]
I got,How we can check the password.Thanks for help

**Comment by pglombardo** [comment]
Where are you hosting the application?

**Comment by Abhishek0237** [comment]
In docker

**Comment by pglombardo** [comment]
> Is it not possible to check the password shared by user from Admin dashboard?

Not possible.  That would lessen the security of the application.  Push content is encrypted and in the database until expiration.

**Comment by Abhishek0237** [comment]
Okay Thanks for the confirmation @pglombardo

**Comment by Abhishek0237** [comment]
One thing more @pglombardo do we enable the 2fa for logins,specially for admin account..

**Comment by pglombardo** [comment]
Hi @Abhishek0237 - 2FA isn't supported yet but it's coming soon!

**Comment by a1010s** [comment]
@pglombardo I enabled logins, all works well (I can login) but there is no Admin Dashboard at /admin.
I get a 404. Any Ideas? Also when trying to start the console, it exits out:

```bash
pwpush-8486c9c487-q6c48:/opt/PasswordPusher/bin$ ./pwpush console
Password Pusher Version: 1.43.0
/usr/local/lib/ruby/3.3.0/logger/log_device.rb:105:in `initialize': No such file or directory @ rb_sysopen - log/production.log (Errno::ENOENT)
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:105:in `open'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:105:in `create_logfile'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:99:in `rescue in open_logfile'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:95:in `open_logfile'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:88:in `set_dev'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:18:in `initialize'
	from /usr/local/lib/ruby/3.3.0/logger.rb:589:in `new'
	from /usr/local/lib/ruby/3.3.0/logger.rb:589:in `initialize'
	from /opt/PasswordPusher/config/environments/production.rb:64:in `new'
	from /opt/PasswordPusher/config/environments/production.rb:64:in `block in <main>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/railtie.rb:257:in `instance_eval'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/railtie.rb:257:in `configure'
	from /opt/PasswordPusher/config/environments/production.rb:5:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/zeitwerk-2.6.16/lib/zeitwerk/kernel.rb:34:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:568:in `block (2 levels) in <class:Engine>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:567:in `each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:567:in `block in <class:Engine>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:32:in `instance_exec'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:32:in `run'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:231:in `block in tsort_each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:353:in `block (2 levels) in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:425:in `block (2 levels) in each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:434:in `each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:424:in `block in each_strongly_connected_component_from'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:50:in `each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:50:in `tsort_each_child'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:418:in `call'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:418:in `each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:352:in `block in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `call'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:229:in `tsort_each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:208:in `tsort_each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:60:in `run_initializers'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/application.rb:426:in `initialize!'
	from /opt/PasswordPusher/config/environment.rb:7:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/zeitwerk-2.6.16/lib/zeitwerk/kernel.rb:34:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/application.rb:402:in `require_environment!'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/actions.rb:20:in `boot_application!'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/commands/console/console_command.rb:105:in `perform'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor/command.rb:28:in `run'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/base.rb:178:in `invoke_command'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor.rb:527:in `dispatch'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/base.rb:73:in `perform'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:71:in `block in invoke'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:149:in `with_argv'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:69:in `invoke'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/commands.rb:18:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from ./pwpush:6:in `<main>'
/usr/local/lib/ruby/3.3.0/logger/log_device.rb:97:in `initialize': No such file or directory @ rb_sysopen - log/production.log (Errno::ENOENT)
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:97:in `open'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:97:in `open_logfile'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:88:in `set_dev'
	from /usr/local/lib/ruby/3.3.0/logger/log_device.rb:18:in `initialize'
	from /usr/local/lib/ruby/3.3.0/logger.rb:589:in `new'
	from /usr/local/lib/ruby/3.3.0/logger.rb:589:in `initialize'
	from /opt/PasswordPusher/config/environments/production.rb:64:in `new'
	from /opt/PasswordPusher/config/environments/production.rb:64:in `block in <main>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/railtie.rb:257:in `instance_eval'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/railtie.rb:257:in `configure'
	from /opt/PasswordPusher/config/environments/production.rb:5:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/zeitwerk-2.6.16/lib/zeitwerk/kernel.rb:34:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:568:in `block (2 levels) in <class:Engine>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:567:in `each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/engine.rb:567:in `block in <class:Engine>'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:32:in `instance_exec'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:32:in `run'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:231:in `block in tsort_each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:353:in `block (2 levels) in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:425:in `block (2 levels) in each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:434:in `each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:424:in `block in each_strongly_connected_component_from'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:50:in `each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:50:in `tsort_each_child'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:418:in `call'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:418:in `each_strongly_connected_component_from'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:352:in `block in each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `call'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:350:in `each_strongly_connected_component'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:229:in `tsort_each'
	from /usr/local/lib/ruby/3.3.0/tsort.rb:208:in `tsort_each'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/initializable.rb:60:in `run_initializers'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/application.rb:426:in `initialize!'
	from /opt/PasswordPusher/config/environment.rb:7:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/zeitwerk-2.6.16/lib/zeitwerk/kernel.rb:34:in `require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/application.rb:402:in `require_environment!'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/actions.rb:20:in `boot_application!'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/commands/console/console_command.rb:105:in `perform'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor/command.rb:28:in `run'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/base.rb:178:in `invoke_command'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/thor-1.3.1/lib/thor.rb:527:in `dispatch'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command/base.rb:73:in `perform'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:71:in `block in invoke'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:149:in `with_argv'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/command.rb:69:in `invoke'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/railties-7.1.3.4/lib/rails/commands.rb:18:in `<main>'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /opt/PasswordPusher/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from ./pwpush:6:in `<main>'
pwpush-8486c9c487-q6c48:/opt/PasswordPusher/bin$
```

**Comment by pglombardo** [comment]
Hi @a1010s - Change the `pwd` to `/opt/PasswordPusher/`.  It's looking for `log/production.log` which is relative and not found under `./bin`.

As for the 404 on /admin - that is by design if you haven't marked your account as administrator.  [These are the docs ](https://docs.pwpush.com/docs/admin-dashboard/#marking-a-user-as-an-administrator)(assuming you haven't found them yet).

**Comment by pglombardo** [comment]
@Abhishek0237 - I'll close this issue out soon.  @a1010s feel free to file an issue if you have any other problems!

**Comment by a1010s** [comment]
@pglombardo Thanks Peter for your prompt answer. I will test like that. I thought so about the 404, that I didn't yet create an admin. 
I will have another question: We would like to implement Azure SSO for Password Pusher. So far, I don't see any option. 
Does this option exist? If not, do you plan to implement it? Will be very happy about this.

**Comment by pglombardo** [comment]
> I will have another question: We would like to implement Azure SSO for Password Pusher. So far, I don't see any option.
> Does this option exist? If not, do you plan to implement it? Will be very happy about this.

This is coming soon.  I've already have the base code and looked into adding a Microsoft SSO option.

Unfortunately I have a backlog of features to get out.  Also 2FA is in the pipeline.

**Comment by pglombardo** [comment]
Just an update that I've updated pwpush.com with SSO for Google and Microsoft.  This is part of a set of premium features I introduced  to support the project.  These eventually get moved to the OSS code base.

For those self-hosting, SSO will be moved to the OSS version soon.  Once out I'll update the feature request in #1609.

For now I'll close out this issue but if anything remains let me know!
